_posts/2020-01-16-spring-bean-life-cycle.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "The Lifecycle of Spring Beans"
+categories: [spring-boot]
+modified: 2020-01-16
+excerpt: "An in-depth look at the paging support provided by Spring Data for querying
+          Spring Web MVC controllers and Spring Data repositories."
+image:
+  auto: 0012-pages
+
+---
+
+## What are Spring Beans?
+ > Spring Beans are Java Objects or Instances which is created and managed by Spring container.
+
+Spring `@ApplicationContext` is responsible for creating a Spring bean and manage the 
+Spring bean life cycle.
+
+Below, We have created a simple Spring bean for `Foo` class using `@Component` annotation.
+
+```java
+@Component
+Class Foo { //variables and methods ...}
+```
+## The Spring Bean Lifecycle
+
+The Bean's life cycle consists of 2 phases `Post-initialization` and `Pre-destruction`. 

```

Comment:
Are `Post-initialization` and `Pre-destruction` (is the capitalization intended?) keywords used by Spring? Highlighting them like this makes the reader think so. Perhaps link to the Spring documentation where this is documented? Also, explain in a few words what post-init and pre-destruction is.